### PR TITLE
task/wg-199-react-query-initial-setup

### PR DESCRIPTION
**Pull request by sophia-massie**
## PR Status: ##

* [X] Ready.
* [ ] Work in Progress.
* [ ] Hold.

## Related Jira tickets: ##

* [WG-199](https://tacc-main.atlassian.net/browse/WG-199)

## Testing Steps: ##
1.

### CODE REVIEW

File: react/src/types/projects.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  ds_id?: string;
+  title?: string;
+  uuid?: string;
+  public?: boolean;
+  system_file?: string;
+  system_id?: string;
+  system_path?: string;
+  deletable?: boolean;
+  deleting?: boolean;
+  deletingFailed?: boolean;
```
Comment (nathanfranklin) [inline react/src/types/projects.ts:14]: I think these two can be dropped.  They aren't part of the response from the backend;  I think they are added on the angular side but won't be needed in react.
```suggestion
```

File: react/src/types/projects.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+export interface Project {
+  description: string;
+  id?: number;
+  name: string;
+  ds_id?: string;
```
Comment (nathanfranklin) [inline react/src/types/projects.ts:5]: nothing to change but just wanted to point out that this isn't a response from the backend but an extra property which was used in the angular client.  no need to change it but just an fyi.

Comment (sophia-massie) [reply react/src/types/projects.ts:5]: I can take this out with the other unneeded properties

File: react/src/hooks/projects/useProjects.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { UseQueryResult } from 'react-query';
+import { Project } from '../../types';
+import { useGet } from '../../requests';
+
+const useProjects = (): UseQueryResult<Project[]> => {
+  const query = useGet<Project[]>({
+    endpoint: '/projects/',
+    key: ['project'],
```
Comment (nathanfranklin) [inline react/src/hooks/projects/useProjects.ts:8]: `projects`?

Comment (sophia-massie) [reply react/src/hooks/projects/useProjects.ts]: Good catch thank you

**Review by nathanfranklin — APPROVED** [review]
LGTM 👍

**Pull request merged**